src/core/algorithms/nar/value_range.cpp

```diff
@@ -0,0 +1,45 @@
+#include "value_range.h"
+
+namespace model {
+
+StringValueRange::StringValueRange(TypedColumnData const& column) {
+    for (size_t row_index = 0; row_index < column.GetNumRows(); ++row_index) {
+        std::byte const* value = column.GetValue(row_index);
+        std::string string_value = Type::GetValue<std::string>(value);
+        bool first_occurrence =
+                std::find(domain.begin(), domain.end(), string_value) == domain.end();
+        if (first_occurrence) {
+            domain.push_back(std::move(string_value));
+        }
+    }
```

Comment:
IMHO smth like this is easier:
```suggestion
    std::unordered_set<std::string> unique_values;
    for (size_t row_index = 0; row_index < column.GetNumRows(); ++row_index) {
        std::byte const* value = column.GetValue(row_index);
        auto string_value = Type::GetValue<std::string>(value);
        if (unique_values.insert(string_value).second) {
            domain.push_back(std::move(string_value));
        }
    }
```